### DISCUSSION TITLE
trying to use GPU instead of CPU for processing

### DISCUSSION THREAD

**Discussion by Ni-Deshmane** [lllyasviel/Fooocus#3556] [Q&A] [answered]
Hi, I'm using a RTX4070, but I can't get Fooocus to use the GPU for processing.  I tried updating the run.bat file I tried with the this code:

set CUDA_VISIBLE_DEVICES=1 

but I am getting RuntimeError: No CUDA GPUs are available. How can I have it choose my Geforce GPU instead of the Intel one?

I am new to this, so please paste the full code, not just a snippet with no context of how/where to use it.  Thank you!

**Comment by Rogala** [comment, marked as answer]
run.bat
```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --gpu-device-id 1
pause
```

**Reply by Ni-Deshmane** [reply]
I'm getting an error that says:
no CUDA GPUs are available

**Reply by Rogala** [reply]
I don't know what number your video card has in the system, open Task Manager and see if it's 0 or 1

**Reply by Ni-Deshmane** [reply]
Hey thanks for your input! I figured it out:

.\python_embeded\python.exe -s Fooocus\entry_with_update.py --gpu-device-id 0
pause  

But I also had to change my Nvidia driver settings to always use the advanced 3D image settings, set 3D settings to 'High Performance Nvidia processor'.  Hope that helps anyone else stuck with this issue

![image](https://github.com/user-attachments/assets/57f3f15c-ab34-459a-a010-de39b8bf84ac)
![image](https://github.com/user-attachments/assets/99bd1f30-f1e3-4662-a85e-012fb2cec1c6)
